### PR TITLE
enable dynamic presets

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: enable dynamic presets

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

```release-note
enable dynamic presets option
```

### CODE REVIEW

**Comment by xrstf** [comment]
Is the old presets.yaml still something the operator should support? If not, can we remove it from the operator code? Otherwise please extend it so the operator can handle the old and new preset stuff.

**Comment by zreigz** [comment]
`presets.yaml` it a static way of loading presets and it's replaced by `loodse-preset.yaml` right now. The new one is deployed here: https://github.com/kubermatic/infra/blob/master/hack/ci/deploy-seeds.sh#L31

**Comment by alvaroaleman** [comment]
/hold

> Is the old presets.yaml still something the operator should support? If not, can we remove it from the operator code? Otherwise please extend it so the operator can handle the old and new preset stuff.

Please extend the operator like you extended the chart.

File: api/pkg/controller/operator/master/resources/kubermatic/api.go [hunk 1]
```diff
@@ -142,6 +142,8 @@ func APIDeploymentCreator(cfg *operatorv1alpha1.KubermaticConfiguration, workerN
 					Name:      "presets",
 					ReadOnly:  true,
 				})
+			} else {
```
Comment (zreigz) [inline api/pkg/controller/operator/master/resources/kubermatic/api.go]: @xrstf is it OK now?

Comment (zreigz) [reply api/pkg/controller/operator/master/resources/kubermatic/api.go]: the operator right now supports static and dynamic presets

**Comment by alvaroaleman** [comment]
I think this feature needs docs now, both for the operator and the chart-based installation.

**Comment by xrstf** [comment]
> presets.yaml it a static way of loading presets and it's replaced by loodse-preset.yaml right now. The new one is deployed here: https://github.com/kubermatic/infra/blob/master/hack/ci/deploy-seeds.sh#L31

@zreigz 

When will the preset CRD replace the old YAML mechanism? I don't want to build something into the operator that is already dead/reprecated. Can we not just simply remove everything preset-related from the operator?

**Comment by zreigz** [comment]
@xrstf PTAL

**Review by xrstf — APPROVED** [review]
Nice :)

**Comment by zreigz** [comment]
@alvaroaleman PTAL

**Comment by alvaroaleman** [comment]
@zreigz is it intended that this now changes the operator only?

**Comment by zreigz** [comment]
@alvaroaleman it supports also the API deployment template to use dynamic presets. The secret repo has been changed to use it. Right now we don't have presets on dev environment without those changes

**Review by alvaroaleman — APPROVED** [review]
/lgtm
/approve

**Comment by zreigz** [comment]
I had to bump chart version, PTAL

**Review by alvaroaleman — APPROVED** [review]
/lgtm
/approve

**Comment by toschneck** [comment]
@zreigz please update the docs as well, presets are already used by customers. We need to update:
https://docs.loodse.com/kubermatic/master/advanced/presets/
and also a note in the changelog

**Comment by zreigz** [comment]
@toschneck done https://github.com/loodse/docs/pull/221

**Pull request merged**